packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -0,0 +1,473 @@
+    return dropAreaWithoutMargin + _kDropAreaMargin;
+  }
+
+  @override 
+  void initState() {
+    super.initState();
+    _entranceController = new AnimationController(vsync: this, value: 0.0, duration: _kReorderAnimationDuration);
+    _ghostController = new AnimationController(vsync: this, value: 0.0, duration: _kReorderAnimationDuration);
+    _entranceController.addStatusListener(_onEntranceStatusChanged);
+  }
+
+  @override
+  void dispose() {
+    _entranceController.dispose();
+    _ghostController.dispose();
+    super.dispose();
+  }
+
+  // Animates the dropable space from _currentIndex to _nextIndex.
+  void _requestAnimationToNextIndex() {
+    if (_entranceController.isCompleted) {
+      _ghostIndex = _currentIndex;
+      if (_nextIndex == _currentIndex) {
+        return;
+      }
+      _currentIndex = _nextIndex;
+      _ghostController.reverse(from: 1.0);
+      _entranceController.forward(from: 0.0);
+    }
+  }
+
+  // Requests animation to the latest next index if it changes during an animation.
+  void _onEntranceStatusChanged(AnimationStatus status) {
+    if (status == AnimationStatus.completed) {
+      setState(() {
+        _requestAnimationToNextIndex();
+      });
+    }
+  }
+
+  // Scrolls to a target context if that context is not on the screen.
+  void _scrollTo(BuildContext context) {
+    if (_scrolling) 
+      return;
+    final RenderObject contextObject = context.findRenderObject();
+    final RenderAbstractViewport viewport = RenderAbstractViewport.of(contextObject);
+    assert(viewport != null);
+    // If and only if the current scroll offset falls in-between the offsets
+    // necessary to reveal the selected context at the top or bottom of the
+    // screen, then it is already on-screen.
+    final double margin = _dropAreaExtent;
+    final double scrollOffset = _scrollController.offset;
+    final double topOffset = max(
+      _scrollController.position.minScrollExtent, 
+      viewport.getOffsetToReveal(contextObject, 0.0).offset - margin,
+    );
+    final double bottomOffset = min(
+      _scrollController.position.maxScrollExtent,
+      viewport.getOffsetToReveal(contextObject, 1.0).offset + margin,
+    );
+    final bool onScreen = scrollOffset <= topOffset && scrollOffset >= bottomOffset;
+    // If the context is off screen, then we request a scroll to make it visible.
+    if (!onScreen) {
+      _scrolling = true;
+      _scrollController.position.animateTo(
+        scrollOffset < bottomOffset ? bottomOffset : topOffset, 
+        duration: _kScrollAnimationDuration, 
+        curve: Curves.easeInOut,
+      ).then((Null none) {
+        setState(() {
+          _scrolling = false;
+        });
+      });
+    }
+  }
+
+  // Wraps children in Row or Column, so that the children flow in
+  // the widget's scrollDirection.
+  Widget _buildContainerForScrollDirection({List<Widget> children}) {
+    if (widget.scrollDirection == Axis.horizontal) {
```

Comment:
switch (widget.scrollDirection)

Reply:
That results in the following lint warning:
>    This function declares a return type of 'Widget', but doesn't end with a return statement.

I think the best resolution is to leave this as-is since it's less code and passes the lint without any special changes. Let me know if you have any objections.